### ISSUE TITLE
to_netcdf uses deprecated and unnecessary dask call

### ISSUE THREAD

**Issue by crusaderky** [pydata/xarray#2273]
```
>>> ds = xarray.Dataset({'x': 1})
>>> ds.to_netcdf('foo.nc')
dask/utils.py:1010: UserWarning: Deprecated, see dask.base.get_scheduler instead
```

Stack trace:
```
> xarray/backends/common.py(44)get_scheduler()
     43         from dask.utils import effective_get
---> 44         actual_get = effective_get(get, collection)
```
There are two separate problems here:

- dask recently changed API from ``get(get=callable)`` to ``get(scheduler=str)``. Should we
   - just increase the minimum version of dask (I doubt anybody will complain)
   - go through the hoops of dynamically invoking a different API depending on the dask version :sweat:
   - silence the warning now, and then increase the minimum version of dask the day that dask removes the old API entirely (risky)?
- xarray is calling dask even when it's unnecessary, as none of the variables in the example Dataset had a dask backend.
I don't think there are any CI suites for NetCDF without dask. I'm also wondering if they would bring any actual added value, as dask is small, has no exotic dependencies, and is pure Python; so I doubt anybody will have problems installing it whatever his setup is.

@shoyer opinion?

**Comment by jhamman** [comment]
@crusaderky - I've been trying to get around to fixing this. @fujiisoup noticed it a few days back: https://github.com/pydata/xarray/issues/2238.

**Comment by NicWayand** [comment]
For anyone else looking for a TEMP fix to hide these warnings (they were spamming my output making debugging difficult)...
`import warnings`
`warnings.simplefilter(action='ignore', category=UserWarning) `

**Comment by jhamman** [comment]
this should have been closed by #2282

**Comment by NicWayand** [comment]
Ah thanks @jhamman!
